### DISCUSSION TITLE
Feedback for major sections of our community strategy document

### DISCUSSION THREAD

**Discussion by choldgraf** [executablebooks/meta#494] [Q&A]
This is a thread for discussion and comments around the major sections of the community strategy document described in [this issue](https://github.com/executablebooks/meta/issues/493).

There is a sub-comment in this Discussion for each of the major areas described in that document. If you wish to have discussion and comments about a specific sub-topic, respond in the corresponding thread to discuss there.

**Comment by choldgraf** [comment]
### mission and values of the organization ([link](https://hackmd.io/8Fd5kEpoSZGlF6c5tNlLuw?both#Mission-and-values-of-the-community))

**Comment by choldgraf** [comment]
### governance and decision-making process ([link](https://hackmd.io/8Fd5kEpoSZGlF6c5tNlLuw?both#Governance))

**Comment by choldgraf** [comment]
### Code of Conduct ([link](https://hackmd.io/8Fd5kEpoSZGlF6c5tNlLuw?both#Code-of-Conduct))

**Comment by choldgraf** [comment]
### Community Practices ([link](https://hackmd.io/8Fd5kEpoSZGlF6c5tNlLuw?both#Community-practices))

**Reply by firasm** [reply]
In this section you have this bit:

> There are (roughly) two kinds of contributors that will interact with our project:
> 
> 1. People who make small contributions, and will move on relatively soon
> 1. People who make small contributions, and continue contributing enough that they become maintainers and co-leaders
> 
> The people in group 1 provide short-term value to the project, though if it takes extensive time to accept their contributions it may be a net-negative expense of energy for the maintainers. The people in group 2 are incredibly valuable for an open source project, they are the only way that it continues to survive as a multi-stakeholder project. The goal of an open community like ours is to create more people in group 2

I wanted to propose a couple of minor adjustments here, mostly for selfish purposes.

1. I think there is a third group of people that exists that may be worth including here:
- People who *want* to help move the project forward but aren't quite sure what is needed, nor where to start.

One subset of this group may have the technical chops to do PRs, but they don't have a good enough sense of the codebase nor the needs of the community to figure out where their contributions would be most welcome.

Another subset of the group may not have the technical chops to do PRs, but they can help triage issues, troubleshoot with new users, and contribute to the overall maintenance and moderation of the issue tracker/discussions etc...With much less overhead than shepherding along a new PR from a new unproven contributor, these people can become vital parts of the project ecosystem and free up the core maintainers/contributors to tackle bigger scope issues.

I suppose what I'm suggesting is that we consider this third tier of people and create some resources/tasks/workflows so they can contribute in the background while they learn the codebase, get a sense of what the community needs are, so they eventually end up (in your) group 1 or 2. 

2. Net benefit 

Though I totally agree that practically, "net expense of energy for the maintainers" is a very important metric for the long term sustainability of the project, I fear that writing it this way may turn people away from contributing minor/small fixes to the project. If someone adds a paragraph explanation to an obscure portion of the documentation that no one reads, and a maintainer has to spend 15-20 minutes reviewing the PR, I would probably classify that as a "net-negative" energy expenditure, even though the project has moved forward (incrementally).

I suppose this is my bias from teaching students coming through, but put simply, I imagine that most external contributions in the eco system will be net negative - core-maintainers devs will **always** be able to do things better, faster, and more expertly with more edge-cases, and future features in mind. That said, having more contributors, users, and "impact" overall is what I would value.

I'm happy to take a stab at rewording this and submitting a PR if people are okay with it, though I admit this is a very minor section of the proposal.

**Reply by chrisjsewell** [reply]
> One subset of this group may have the technical chops to do PRs, but they don't have a good enough sense of the codebase nor the needs of the community to figure out where their contributions would be most welcome.

Heya, just some quick thoughts on this:

For the needs of the community, I guess a good starting point would be: https://executablebooks.org/en/latest/feature-vote.html

Then also for figuring out what to contribute there are a few "good first issue" tags spread around, e.g.

- https://github.com/executablebooks/jupyter-book/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+issue%22
- https://github.com/executablebooks/MyST-Parser/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+issue%22

Perhaps more issues should be tagged with this (and include and decent description of what needs to be done), and also they should be "publisised" more?

**Reply by chrisjsewell** [reply]
> If someone adds a paragraph explanation to an obscure portion of the documentation that no one reads, and a maintainer has to spend 15-20 minutes reviewing the PR, I would probably classify that as a "net-negative" energy expenditure

Personally, as a maintainer, I would say these documentation PRs are the best thing we can get from new contributors: they rarely take more than about a minute to review, because (1) they usually come from the experience of a user having "figured" something out, and (2) I know there is not really any risk of merging them, as opposed to code PRs which intrinsically have the potential to break the whole application.

Especially given sphinx-book-theme's nice button for opening a PR, we get a lot of these: https://github.com/executablebooks/jupyter-book/pulls?q=is%3Apr+sort%3Aupdated-desc+is%3Aclosed+DOCS+in%3Atitle, and they always feel like a quick win to engage with the community

**Reply by choldgraf** [reply]
@firasm thanks for these points, I agree with the added nuance that you're suggesting here. I'm happy to look at some suggested edits! Mozilla has this concept of the 'Mountain of Engagement' for projects, which basically says that the large majority of people will be only marginally engaged, and as you go up the "engagement scale", the number of people become fewer and fewer. However, everybody has to start somewhere, and most people start at the bottom. So the trick is building pathways for people to go from user -> casual contributor -> regular contributor -> maintainer, and also to do this along a few different dimensions other than just code development. I'll try and look up some of the Mozilla docs and see what I can find. EDIT: here is one post about it: https://www.openscapes.org/blog/2021/01/06/audience-building/

Also agree w/ @chrisjsewell that documentation improvements are a really helpful way for users to contribute - this is especially true because users have a perspective that maintainers don't have - they are not as familiar with the tool in the first place!

**Reply by foster999** [reply]
I think the project has done a particularly good job of reaching out to technical users and maintainers (through feature voting and these Discussions), but feel like there might be a gap in the accessibility for non-technical users to feedback or contribute?

I'm thinking of end-users reading executable books, who may not have the coding background or GitHub account to make contributions in these ways. [The Turing Way](https://the-turing-way.netlify.app/community-handbook/community-handbook.html) for example provides points of contact via Google Forms or email, which are accessible to anyone. I'm not sure how many would want to engage in this way, but it could be supported by splitting contribution guidance by the personas of a few anticipated types of contributor.

**Reply by choldgraf** [reply]
@foster999 I think that's a great point - we have some unofficial contact points but mostly contact is via the GitHub issues and discussion board. The challenge with adding new communication channels is ensuring that we can sustain the effort needed to pay attention to them etc, but at the least having _some_ ability for people to contact us outside of GitHub is a good idea. At the very least, this is a necessity in the case of Code of Conduct violations.

(and more generally, I think that we should basically just try and do whatever the Turing Way community does - that is a wonderful example of an inviting community :-) )

**Reply by chrisjsewell** [reply]
> At the very least, this is a necessity in the case of Code of Conduct violations.

We do already have this @ebp-bot has executablebooks@gmail.com, which is specified in https://github.com/executablebooks/.github/blob/d55b5e5b65a984c4159de39b04fa8852aeae0315/CODE_OF_CONDUCT.md#enforcement

**Reply by firasm** [reply]
Thanks @chrisjsewell and @choldgraf ! By the way, I certainly didn't mean any of what I said to be a criticism, just suggestions for improving engagement with average users. 

I like the `good-first-issue` ; some other tags could be useful as well like, `support-request` or `question`, etc...

And really good to hear about the documentation improvements being welcome, when I have some time after the term is done, I'll go through and add some contributions at places where I thought things were confusing or hard to parse.

**Comment by choldgraf** [comment]
### Organizational designation ([link](https://hackmd.io/8Fd5kEpoSZGlF6c5tNlLuw?both#Long-term-organizational-home))
